File: PropertyAPI/PropertyAPI.Domain/Entities/Property.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace PropertyAPI.Domain.Entities;

public class Property
{
  [BsonId]
  [BsonRepresentation(BsonType.ObjectId)]
  public string? Id { get; set; }

  [BsonElement("idOwner")]
  public required string IdOwner { get; set; }

  [BsonElement("name")]
  public required string Name { get; set; }

  [BsonElement("address")]
  public required string Address { get; set; }

  [BsonElement("price")]
  public decimal Price { get; set; }

  [BsonElement("imageUrl")]
  public string? ImageUrl { get; set; }

  [BsonElement("createdAt")]
  public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend-react/src/types/property.ts
export interface Property {
  id?: string;
  idOwner: string;
  name: string;
  address: string;
  price: number;
  imageUrl?: string;
  createdAt?: Date;
}

export interface PropertyFilter {
  name?: string;
  address?: string;
  minPrice?: number;
  maxPrice?: number;
}

export interface CreatePropertyData {
  idOwner: string;
  name: string;
  address: string;
  price: number;
  imageUrl?: string;
}
